middle_end/flambda/terms/flambda_unit.ml

```diff
@@ -59,7 +60,56 @@ let print ppf
 
 let invariant _t = ()
 
-let used_closure_vars t = t.used_closure_vars
+let apply_name_permutation { return_continuation; exn_continuation;
+                             body; module_symbol; used_closure_vars; } perm =
+  let body = Expr.apply_name_permutation body perm in
+  let module_symbol = Name_permutation.apply_symbol perm module_symbol in
+  let return_continuation = Name_permutation.apply_continuation perm return_continuation in
+  let exn_continuation = Name_permutation.apply_continuation perm exn_continuation in
+  let used_closure_vars = Or_unknown.map used_closure_vars ~f:(fun set ->
+    Var_within_closure.Set.map (fun var ->
+      let comp_unit = Var_within_closure.get_compilation_unit var in
+      let v = Var_within_closure.unwrap var in
+      let v' = Name_permutation.apply_variable perm v in
+      Var_within_closure.wrap comp_unit v'
+    ) set)
```

Comment:
This looks wrong. `Var_within_closure.unwrap` creates a new variable, as there's no variable associated to a `Var_within_closure.t`. So the new variable won't be part of the permutation, as it's guaranteed to be fresh.
There's no ID (in the sense of `Ids_for_export`) associated to `Var_within_closure.t`s anyway, so the `permute_everything` function below would not permute them.

Reply:
So actually, we should ignore the `Var_within_closure.t`s and not try and permute them when permuting everything, is that it ?

Reply:
Yes. We can't permute them anyway at the moment. You may want to leave a comment explaining this though. (Also, closure variables behave a bit like symbols and code IDs: you can permute the ones from your own compilation unit, but you must not change the ones coming from external units. Since you do the permutation before inlining at the moment then you will not encounter variables from external units, but after inlining this could happen).

Reply:
The part about `var_within_closure` has been removed